oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java

```diff
@@ -3405,9 +3405,18 @@ private String diffImpl(AbstractDocumentNodeState from, AbstractDocumentNodeStat
                 fromRev = from.getRootRevision();
                 toRev = to.getRootRevision();
             } catch (RuntimeException e) {
-                LOG.warn("diffJournalChildren failed with " +
-                        e.getClass().getSimpleName() +
-                        ", falling back to classic diff", e);
+                // avoid filling the log file with stack traces for a known issue
+                // see OAK-6016 and OAK-6011
+                if (e instanceof IllegalStateException &&
+                        "Root document does not have a lastRev entry for local clusterId 0".equals(e.getMessage())) {
+                    LOG.warn("diffJournalChildren failed with " +
+                            e.getClass().getSimpleName() +
+                            ", falling back to classic diff");
```

Comment:
```suggestion
                            ", falling back to classic diff : " + e.getMessage());
```
minor comment : could consider re-adding the message as that would be slightly more explicit